### ISSUE TITLE
list metrics files only

### ISSUE THREAD

**Issue by gennaro-tedesco** [iterative/dvc#7133]
With `dvc list -R --dvc-only .` one can obtain all dvc-committed files in the current working directory. It could be useful to introduce a similar command that helps differentiating files from metrics (and more generally outputs or dependencies): this is because very often - when having many different metrics files - you want to list only one type and have a quick look at this metrics only.

At the moment I am doing it quite hackily with
```
dvc metrics show --json | jq '."".data' | grep json | cut -d'"' -f2
```
using the fact that `dvc metrics show` lists the names of the files, so I parse them into an independent list.

## Suggestion

Why not creating some flags like 
```
dvc list -m  <--- for metrics
dvc list -o  <--- for outputs
dvc list -d  <--- for dependencies
```
and so forth?

P. S. Could you point me at where exactly in the code these commands are defined? I might try to have a look myself and initiate a skeleton PR if need be.

**Comment by pared** [comment]
Original discussion:
https://discord.com/channels/485586884165107732/485596304961962003/919899944213680129

So the `list` itself is defined in `dvc/command/ls/__init__.py`. Is that what you looked for? If you have more questions feel free to ping me :)

**Comment by daavoo** [comment]
Do you want to do this directly from CLI or would you be able to use Python??

There is the internal `Repo` API (not documented yet) with an `index` object (defined in https://github.com/iterative/dvc/blob/master/dvc/repo/index.py) that can give you access to pretty much everything tracked:

```python
from dvc.api import Repo

repo = Repo(".")

# dvc list -m
 list(repo.index.metrics)
 # dvc list -o 
 list(repo.index.outs)
# dvc list -d
 list(repo.index.deps)
```

**Comment by gennaro-tedesco** [comment]
Ideally from command line, since these are part of other shell functions. Do I understand correctly that one may just expose the internal `Repo` API `from dvc.api import Repo` in [here](https://github.com/iterative/dvc/blob/602713e57bf846a7aeede7c5b337d35d1e55e57f/dvc/repo/ls.py#L49)?

**Comment by daavoo** [comment]
> Do I understand correctly that one may just expose the internal `Repo` API `from dvc.api import Repo` in [here](https://github.com/iterative/dvc/blob/602713e57bf846a7aeede7c5b337d35d1e55e57f/dvc/repo/ls.py#L49)?

Yes. All DVC commands are pretty much a wrapper of `Repo` API calls with some additions like return codes and, in some cases, some custom logic after the `Repo` API call.

**Comment by skshetry** [comment]
@daavoo @gennaro-tedesco, note that `Repo` is not a part of `dvc.api`, which is an official/public API of DVC. `Repo` is an internal API and can be imported as:

```python
from dvc.repo import Repo
```

**Comment by daavoo** [comment]
> @daavoo @gennaro-tedesco, note that `Repo` is not a part of `dvc.api`, which is an official/public API of DVC. `Repo` is an internal API and can be imported as:
> 
> ```python
> from dvc.repo import Repo
> ```

@skshetry It is actually the same, right?:

```
>>> from dvc.api import Repo
>>> Repo
<class 'dvc.repo.Repo'>
>>> from dvc.repo import Repo
>>> Repo
<class 'dvc.repo.Repo'>
```

**Comment by skshetry** [comment]
@daavoo, yes but `dvc.repo.Repo` is a canonical path for it. Mentioning it as `dvc.api.Repo` makes it sound like it's a part of `dvc.api` which it is not. It is a side-effect of an import in `dvc.api` due to Python's implicit exports.
